app/models/host/base.rb

```diff
@@ -102,6 +102,7 @@ def initialize(*args)
       super(*args)
 
       build_required_interfaces
+      self.name ||= Haikunator.haikunate if Setting['default_hostname']
```

Comment:
Warning: There is N : 16777216 chance to return existing name (where N is number of existing and generated foreman hosts and the second number can be calculated from the codebase - 4096*64*64). You should check for existing names and fetch another one in case it's occupied.